cmake/fmtlib.cmake

```diff
@@ -1,20 +1 @@
-include(FetchContent)
-
-FetchContent_Declare(
-	fmtlib
-	PREFIX "${PROJECT_BINARY_DIR}/deps"
-	DOWNLOAD_DIR "${PROJECT_SOURCE_DIR}/deps/downloads"
-	DOWNLOAD_NAME fmt-9.1.0.tar.gz
-	URL https://github.com/fmtlib/fmt/archive/9.1.0.tar.gz
-	URL_HASH SHA256=5dea48d1fcddc3ec571ce2058e13910a0d4a6bab4cc09a809d8b1dd1c88ae6f2
-)
-
-if (CMAKE_VERSION VERSION_LESS "3.14.0")
-	FetchContent_GetProperties(fmtlib)
-	if (NOT fmtlib_POPULATED)
-		FetchContent_Populate(fmtlib)
-		add_subdirectory(${fmtlib_SOURCE_DIR} ${fmtlib_BINARY_DIR})
-	endif()
-else()
-	FetchContent_MakeAvailable(fmtlib)
-endif()
+add_subdirectory(${CMAKE_SOURCE_DIR}/deps/fmt)
```

Comment:
you could add a check whether the submodule(s) have been initialized and otherwise issue a cmake config error

Reply:
@aarlt What about this one? They way you have it now (i.e. having a *non-empty* check for every module) probably solves the problem of supporting the addition of new modules (assuming that `git submodule update --init` will not fail when some submodules are initialized and some are not), but I still think that initializing each submodule in its `.cmake` file would be a better way to organize things. It would put everything related to a specific dependency together and also we would not need to repeat the `USE_SYSTEM_LIBRARIES` check.